client/components/LowerCarousel.jsx

```diff
@@ -1,15 +1,88 @@
+/* eslint-disable no-param-reassign */
+/* eslint-disable react/prop-types */
 /* eslint-disable no-console */
 import React from 'react';
+import styled from 'styled-components';
 import Thumbnail from './Thumbnail';
 import Button from './Button';
 
+const GridWrapper = styled.div`
+  width: 500px;
+`;
+const GridContainer = styled.div`
+  display: grid;
+  grid-template-columns: 48px auto 48px;
+  grid-template-rows: 100%;
+  margin: 0 auto;
+  width: 500px;
+`;
+
+const CarouselWrapper = styled.div`
+  padding: 12px 4px;
+  width: ${(props) => props.numOfThumbnails * 70}px;
+  overflow: hidden;
+`;
+
+const InnerCarousel = styled.div`
+  display: flex;
+  flex-direction: row;
+  align-items: center;
+  position: relative;
+  left: ${(props) => props.position}px;
+  width: 100%;
+`;
+
 class LowerCarousel extends React.Component {
   constructor(props) {
     super(props);
-    this.carouselLoader = this.carouselLoader.bind(this);
+    this.state = {
+      firstVisibleThumbnail: 0,
+      // Calculates number of thumbnails that will fit evenly into the carousel width
+      // TODO: make this generate dynamically based on width of parent div
+      numOfThumbnails: Math.floor((500 - 48 * 2 - 8) / 70),
+      carouselPosition: 0,
+    };
+    this.thumbnailLoader = this.thumbnailLoader.bind(this);
   }
 
-  carouselLoader() {
+  componentDidUpdate() {
+    const { activeThumbnail, images } = this.props;
+    const { firstVisibleThumbnail, numOfThumbnails } = this.state;
+
+    // checks if new selected Thumbnail is currently hidden to the right
+    let isHidden = activeThumbnail > firstVisibleThumbnail + numOfThumbnails - 1
```

Comment:
You could add parenthesis for clarity of checks?

Reply:
Thanks Matt. I like your suggestions. I'll incorporate them all in the next feature branch. I've already made a commit there.